server/core/runtime/plan_step_runner.go

```diff
@@ -273,6 +273,14 @@ func StripRefreshingFromPlanOutput(output string, tfVersion *version.Version) st
 	return output
 }
 
+func FilterRegexFromPlanOutput(output string, filterRegex *regexp.Regexp) string {
+	if filterRegex == nil {
+		return output
+	}
+
+	return filterRegex.ReplaceAllString(output, "${1}<redacted>$2")
```

Comment:
It seems like the logic here is that we keep the first and second capture group, if they exist, and put them before and after `<redacted>`. I don't see this in the documentation, can this be added?

What happens if there are zero, one, or more than two capture groups? We should have test cases in `TestCustomRegexFromPlanOutputFromPlanOutput` to show the expectations here.

Reply:
It will just match everything and return it without any change.

Also if it helps I have been running this in production since I made the PR :) And also already deployed this latest version and it's working fine.

Reply:
I still think this should be documented and tested. I personally don't read `FilterRegexFromPlanOutput` and can tell that's what it's going to do. I can write some tests now that I think I understand, I'm not trying to make busywork for anyone, I just think it would help follow along.

Reply:
Here are some tests:
```
		{
			"foobar",
			"<redacted>oobar",
			"f",
		},
		{
			"foobar",
			"f<redacted>oobar",
			"(f)",
		},
		{
			"foobar",
			"f<redacted>bar",
			"(f)oo(bar)",
		},
```

Reply:
Thank you! I do think tests are important. Adding them.